packages/preferences/src/browser/ui/settings-ui-plus-navbar.tsx

```diff
@@ -0,0 +1,55 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as React from 'react';
+
+interface ItemType {
+    name: string;
+    id: string;
+    [key: string]: any; // eslint-disable-line
+}
+
+function scrollToEditorElement(id: string): boolean {
+    const el = document.getElementById(`${id}-editor`);
+    if (el) {
+        el.scrollIntoView({ behavior: 'auto' });
+        return true;
+    }
+    return false;
+}
+
+export function SettingsUIPlusNavbar({ items }: { items: ItemType[] }): React.ReactElement {
+    return (
+        <ul className="settings-ui-plus-navbar">
+            {items.map(category => (
+                <li
+                    className="settings-ui-plus-navbar-item"
+                    key={`settings-ui-plus-navbar-item-${category.id}`}
+                >
+                    <div
+                        id={category.id}
+                        onClick={(): boolean => scrollToEditorElement(category.id)}
```

Comment:
please review react section in our coding guidelines, here new listener handler is created on each render invalidating reach caches

Reply:
All anonymous arrow function handlers should be refactored now.